### ISSUE TITLE
TCH001 like rule for pre 3.12 generics like TypVar, ParamSpec and the likes

### ISSUE THREAD

**Issue by bvolkmer** [astral-sh/ruff#14434]
It would be nice to have a rule, that checks if a TypeVar, ParamSpec and the likes is defined outside `if TYPE_CHECKING` eventhough it is not needed. Example:
```python
from __future__ import annotations
from typing import TypeVar
_T = TypeVar("_T") # Warning here: Move generics TypeVar into a type-checking block

def foobar(arg: _T):
    return arg
```

could be:

```python
from __future__ import annotations
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import TypeVar

    _T = TypeVar("_T")

def foobar(arg: _T):
    return arg
```

**Comment by Daverball** [comment]
There is a `TCH004` equivalent rule in the upstream `flake8-type-checking` plugin (namely `TC009`). I considered adding a `TCH001-003` equivalent as well, but I deemed its potential for false positives too high. The reverse case is a lot more useful and I will definitely port this rule to ruff.

A more narrow equivalent to `TCH001-003` just for type-vars might make some sense.

Here's a more general discussion about the future of `flake8-type-checking` within ruff: #13208
